### PR TITLE
Sort and refactor methods

### PR DESCRIPTION
**Pull request by drmarjanovic**

### CODE REVIEW

File: modules/library/src/main/scala/zio/elasticsearch/query/FunctionScoreFunction.scala [hunk 1]
```diff
@@ -311,42 +312,42 @@ object FunctionScoreFunction {
...
-      seedAndField = Some(SeedAndField(seed = seed, fieldName = field)),
+      seedAndField = Some(SeedAndField(seed = seed, fieldName = field.toString)),
       weight = None
     )
 
   /**
    * Constructs an instance of [[zio.elasticsearch.query.RandomScoreFunction]] using the specified parameters.
    * [[zio.elasticsearch.query.RandomScoreFunction]] generates scores that are uniformly distributed from 0 up to but
-   * not including 1. By default, it uses the internal Lucene doc IDs as a source of randomness, which is very efficient
+   * not including 1. By default, it uses the internal Lucene doc ids as a source of randomness, which is very efficient
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/query/FunctionScoreFunction.scala:336]:
```suggestion
   * not including 1. By default, it uses the internal Lucene doc IDs as a source of randomness, which is very efficient
```

File: modules/library/src/main/scala/zio/elasticsearch/query/FunctionScoreFunction.scala [hunk 2]
```diff
@@ -536,6 +537,38 @@ final case class FieldValueFactor[S](
...
 
+  def filter[S1 <: S: Schema](filter: ElasticQuery[S1]): FieldValueFactor[S1] =
+    FieldValueFactor[S1](
+      field = self.field,
+      factor = self.factor,
+      filter = Some(filter),
+      modifier = self.modifier,
+      missing = self.missing,
+      weight = self.weight
+    )
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/query/FunctionScoreFunction.scala:548]: Could you use `self.copy(filter = Some(filter))`?

File: modules/library/src/main/scala/zio/elasticsearch/query/FunctionScoreFunction.scala [hunk 3]
```diff
@@ -536,6 +537,38 @@ final case class FieldValueFactor[S](
...
+
+  def filter(value: ElasticQuery[Any]): FieldValueFactor[Any] =
+    FieldValueFactor(
+      field = self.field,
+      factor = self.factor,
+      filter = Some(value),
+      modifier = self.modifier,
+      missing = self.missing,
+      weight = self.weight
+    )
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/query/FunctionScoreFunction.scala:558]: Here also.

File: modules/library/src/main/scala/zio/elasticsearch/query/FunctionScoreFunction.scala [hunk 4]
```diff
@@ -627,6 +628,12 @@ private[elasticsearch] final case class RandomScoreFunction[S](
   weight: Option[Double]
 ) extends FunctionScoreFunction[S] { self =>
 
+  def filter[S1 <: S: Schema](filter: ElasticQuery[S1]): RandomScoreFunction[S1] =
+    RandomScoreFunction[S1](Some(filter), seedAndField, weight)
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/query/FunctionScoreFunction.scala:632]: Name parameters.

File: modules/library/src/main/scala/zio/elasticsearch/query/InnerHits.scala [hunk 5]
```diff
@@ -135,12 +135,10 @@ final case class InnerHits private[elasticsearch] (
   private[elasticsearch] def toStringJsonPair(fieldPath: Option[String]): (String, Json) = {
     val sourceJson: Option[Json] =
       (included, excluded) match {
-        case (Chunk(), Chunk()) =>
-          None
-        case (included, excluded) =>
-          val includes = if (included.isEmpty) Obj() else Obj("includes" -> Arr(included.map(_.toJson)))
-          val excludes = if (excluded.isEmpty) Obj() else Obj("excludes" -> Arr(excluded.map(_.toJson)))
-          Some(includes merge excludes)
+        case (Chunk(), Chunk()) => None
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/query/InnerHits.scala:138]: I guess we can put `None` in the new line, just to be consistent.

**Review by dbulaja98 — CHANGES_REQUESTED** [review]

File: modules/library/src/main/scala/zio/elasticsearch/query/InnerHits.scala (hunk 5 above)
Comment (drmarjanovic) [reply modules/library/src/main/scala/zio/elasticsearch/query/InnerHits.scala:138]: We are. Everything is inlined here.

**Review by dbulaja98 — APPROVED** [review]

**Review by mvelimir — APPROVED** [review]

**Pull request merged**